src/main/java/Duke.java

```diff
@@ -1,10 +1,129 @@
+        } else {
+            newTask = new Event(extractTaskDescription, extractTaskDeadlineTime);
+        }
+
+        taskLists[Task.getNumberOfTasks() - 1] = newTask;
+        System.out.println("Task added:\n\t" + newTask.toString());
+        System.out.println("Now you have " + Task.getNumberOfTasks() + " tasks in your list!");
+    }
+
+    public static boolean isWithinTaskRange(int taskNumber) {
+        if (taskNumber > Task.getNumberOfTasks() || taskNumber <= 0) {
+            System.out.println("Task does not exist!");
+            return false;
+        }
+        return true;
+    }
+
+    public static void markTask(boolean isMarked, String userInput) {
+        int taskNumber = Integer.parseInt(userInput.split(" ")[1]);
+        if (isWithinTaskRange(taskNumber)) {
+            if (isMarked) {
+                taskLists[taskNumber - 1].markAsDone();
+                System.out.println("Fantastic! This task is done:");
+            } else {
+                taskLists[taskNumber - 1].markAsUndone();
+                System.out.println("Uh oh! This task is undone:");
+            }
+            System.out.println(taskLists[taskNumber - 1].toString());
+        }
+    }
+
+    private static void printExit() {
+        System.out.println(BORDER_DECORATION);
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println(BORDER_DECORATION);
+    }
+
+    public static void executeCommand() {
+        String userInput;
+        Scanner in = new Scanner(System.in);
+        userInput = in.nextLine();
+
+        while (!userInput.equalsIgnoreCase("bye")) {
+            System.out.println(BORDER_DECORATION);
+            String command = userInput.split(" ")[0];
+
+            switch (command) {
+            case "list":
+                printList();
+                break;
+            case "todo":
+                addTask(userInput);
+                break;
+            case "deadline":
+                addTaskWithTime(userInput, "/by ");
+                break;
+            case "event":
+                addTaskWithTime(userInput,"/at ");
+                break;
+            case "unmark":
+                markTask(false, userInput);
+                break;
+            case "mark":
+                markTask(true, userInput);
+                break;
+            default:
+                System.out.println("Invalid command.");
+            }
+            System.out.println(BORDER_DECORATION);
+            userInput = in.nextLine();
+        }
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        displayGreeting();
+        executeCommand();
+        printExit();
```

Comment:
personally, I like to put the main method into its own class (for example, a `Main` class), so it is clear where the program actually starts running from.

then, this class will just be dealing with the actual logic required.